monkey/tests/data_for_tests/agent_plugin/mock_multiple_vendors/src/plugin.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import random
+import time
+from threading import Event, current_thread
+from typing import Any, Dict, Sequence
+
+import mock_dependency
+
+from common.agent_events import ExploitationEvent, PropagationEvent
+from common.event_queue import IAgentEventPublisher
+from common.types import AgentID
+from infection_monkey.exploit import IAgentBinaryRepository
+from infection_monkey.i_puppet import ExploiterResultData
+from infection_monkey.model import TargetHost
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+from infection_monkey.utils.threading import interruptible_iter
+
+logger = logging.getLogger(__name__)
+
+
+class Plugin:
```

Comment:
I think we can strip most of the code out of this. In fact, I'd be fine if all it did was log the dependency version and emit/return a failed exploitation event. It doesn't need to duplicate all of the behavior of the other mock plugins.